magda-content-api/src/createApiRouter.ts

```diff
@@ -15,10 +21,138 @@ export default function createApiRouter(options: ApiRouterOptions) {
 
     const router: express.Router = express.Router();
 
+    const ADMIN = mustBeAdmin(options.authApiUrl, options.jwtSecret);
+
     router.get("/healthz", function(req, res, next) {
         res.status(200).send("OK");
     });
 
+    function parseUrlPath(pathString: string) {
```

Comment:
Is /commonAssets necessary?. Custom assets can be configured by use route option. See csv example in content.ts. With that, separate route can be used for different types of resources as well.

Reply:
😄 the `/commonAssets` is not for enforcing a particular route. It's more for providing a little bit different API without impact existing functions.
The logic behind `/commonAssets/*` are aiming for more generic resource storing purposes. The differences are:
- Both GET / POST are using the same resource name.
  - Existing API you need to GET `style.css` but post to `style`
- Handle any type of file without modifying content API or config
- Allow `/` in the resource name. So I reference `emailsTpl/assets/a.png` in email templates

Reply:
- Just want to make it less job to do when uploading content. It will save some code every time you want to upload to content API. Auto figure it out from the extension name could minimize the chance of sending the wrong mime-type
- I agree. It's just a randomly picked name. We probably need a better name 😄 Here just try to make things auto handled and minimize the work when applying the changes. If we want to go for the more configurable direction, we then probably needs some flexibility. However, we probably do need to have some plan with naming.

Reply:
Yeah this sounds more like functionality that should be added to the regular API but I don't think I have the full picture, I'll ask Jacky after lunch 🤔